test/e2e/harness.go

```diff
@@ -94,11 +97,34 @@ func (h *Harness) DeployNativeContract(from *keys.Ed25519KeyPair, contractName s
 }
 
 func (h *Harness) SendTransaction(senderPublicKey []byte, senderPrivateKey []byte, contractName string, methodName string, args ...interface{}) (*codec.TransactionResponse, string, error) {
-	payload, txId, err := h.client.CreateTransaction(senderPublicKey, senderPrivateKey, contractName, methodName, args...)
+	return h.SendTransactionWithClient(h.client, senderPublicKey, senderPrivateKey, contractName, methodName, args...)
+}
+
+func (h *Harness) SendTransactionWithClient(client *orbsClient.OrbsClient, senderPublicKey []byte, senderPrivateKey []byte, contractName string, methodName string, args ...interface{}) (*codec.TransactionResponse, string, error) {
```

Comment:
this method and it's async brother are not really part of the harness. they don't abstract away anything - they are only convenience methods to make two API calls and check the errors. 

consider passing an index of the node you want to send the message to - this way the test code will never have to mess with initializing clients  - it will not have to deal with this plumbing and just tell the harness which node to send to. could make the test code simpler. not a problem as it is just slightly misses the potential of the harness to make the test simpler

Reply:
You can't use the index of the node we did not run. It's an external test.